src/memoize.ts

```diff
@@ -0,0 +1,11 @@
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+export function memoize1<T>(compute: (...rest: any[]) => T) {
```

Comment:
Instead of using `any` for the rest arguments, you can use `unknown`. The difference is subtle, but TS will assume anything about an `any` type, but it won't assume anything about `unknown`.

Reply:
This sounds like very arcane knowledge, black belt! I'm still a yellow belt.

Reply:
Here's a source that talks a bit about the difference: https://www.typescript-training.com/course/fundamentals-v3/11-top-and-bottom-types/

Some useful quotes:

> We can see here that any is not always a “bug” or a “problem” — it just indicates maximal flexibility and the absence of type checking validation.

`any` is the way to tell Typescript that you don't know anything about the type, and that you want to back to the JS rules for types. It's useful, but anytime you do it you are weakening the type of the thing you're working with.

> Values with an unknown type cannot be used without first applying a type guard

`unknown` is the way to tell Typescript that you don't know anything about the type, and that you would like it's help to prove that what you are doing is type safe. It's useful because you are asking Typescript to help make the types stronger.